docroot/modules/custom/mass_translations/mass_translations.module

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/**
+ * @file
+ * Contains mass_translations.module.
+ */
+
+use Drupal\Core\Cache\CacheableMetadata;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Language\LanguageInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Url;
+use Drupal\node\NodeInterface;
+use Drupal\mass_translations\Controller\TranslationsController;
+
+/**
+ * Implements hook_page_attachments().
+ *
+ * Adapted from content_translation.module.
+ */
+function mass_translations_page_attachments(array &$attachments) {
+  $cache = CacheableMetadata::createFromRenderArray($attachments);
+  $route_match = \Drupal::routeMatch();
+
+  // If the current route has no parameters, return.
+  if (!($route = $route_match->getRouteObject()) || !($parameters = $route->getOption('parameters'))) {
+    return;
+  }
+
+  // Determine if the current route represents an entity.
+  foreach ($parameters as $name => $options) {
+    if (!isset($options['type']) || strpos($options['type'], 'entity:') !== 0) {
+      continue;
+    }
+
+    $node_storage = \Drupal::service('entity.manager')->getStorage('node');
+    $translations_controller = new TranslationsController($node_storage);
+
+    $entity = $route_match->getParameter($name);
+    if ($entity instanceof NodeInterface && $entity->hasLinkTemplate('canonical')) {
+      // Current route represents a content entity. Build hreflang links.
+      foreach ($translations_controller->getTranslationLanguages($entity, $node_storage, 'field_english_version') as $language_id => $translation) {
+        // Skip any translation that cannot be viewed.
+        $access = $translation->access('view', NULL, TRUE);
+        $cache->addCacheableDependency($access);
+        if (!$access->isAllowed()) {
+          continue;
+        }
+        $url = $translation->toUrl('canonical')
+          ->setOption('language', $translation->language())
+          ->setAbsolute()
+          ->toString();
+        $attachments['#attached']['html_head_link'][] = [
+          [
+            'rel' => 'alternate',
+            'hreflang' => $language_id,
+            'href' => $url,
+          ],
+          TRUE,
+        ];
+      }
+    }
+    // Since entity was found, no need to iterate further.
+    break;
+  }
+  // Apply updated caching information.
+  $cache->applyTo($attachments);
+}
+
+/**
+ * Implements hook_field_widget_multivalue_form_alter().
```

Comment:
Could we say what we are doing here and why? I'm not sure now, and this will even less clear in the future.

Reply:
Good point. This hides the "Use existing media" text that shows up by default, and I added a note.